ta4j-core/src/main/java/org/ta4j/core/indicators/RandomWalkIndexHighIndicator.java

```diff
@@ -54,14 +52,28 @@ public RandomWalkIndexHighIndicator(TimeSeries series, int timeFrame) {
         this.timeFrame = timeFrame;
         maxPrice = new MaxPriceIndicator(series);
         minPrice = new MinPriceIndicator(series);
-        averageTrueRange = new AverageTrueRangeIndicator(series, timeFrame);
-        sqrtTimeFrame = Decimal.valueOf(timeFrame).sqrt();
+
     }
 
     @Override
     protected Decimal calculate(int index) {
-        return maxPrice.getValue(index).minus(minPrice.getValue(Math.max(0, index - timeFrame)))
-                .dividedBy(averageTrueRange.getValue(index).multipliedBy(sqrtTimeFrame));
+        int lastIndex = Math.max(0, index - timeFrame+1);
+        int n = 2;
+        AverageTrueRangeIndicator averageTrueRange = new AverageTrueRangeIndicator(getTimeSeries(), n);
```

Comment:
Maybe you can extract this to a method instead of duplicating here and in the loop:

```
private Decimal calculateRwiHigh(int index, int pos){
		AverageTrueRangeIndicator atr = new AverageTrueRangeIndicator(getTimeSeries(), pos);
		return maxPrice.getValue(index).minus(minPrice.getValue(Math.max(0, index - 1)))
				.dividedBy(atr.getValue(index).multipliedBy(Decimal.valueOf(Math.sqrt(pos))));
	}
```

The same goes for RWILowInciator.java